src/action/common/place_nix_configuration.rs

```diff
@@ -28,65 +28,191 @@ impl PlaceNixConfiguration {
         extra_conf: Vec<String>,
         force: bool,
     ) -> Result<StatefulAction<Self>, ActionError> {
-        let extra_conf = extra_conf.join("\n");
-        let mut nix_config = nix_config_parser::NixConfig::parse_string(extra_conf, None)
-            .map_err(CreateOrMergeNixConfigError::ParseNixConfig)
+        let create_directory = CreateDirectory::plan(NIX_CONF_FOLDER, None, None, 0o0755, force)
+            .await
             .map_err(Self::error)?;
-        let settings = nix_config.settings_mut();
-
-        settings.insert("build-users-group".to_string(), nix_build_group_name);
-        let experimental_features = ["nix-command", "flakes", "auto-allocate-uids"];
-        match settings.entry("experimental-features".to_string()) {
-            Entry::Occupied(mut slot) => {
-                let slot_mut = slot.get_mut();
-                for experimental_feature in experimental_features {
-                    if !slot_mut.contains(experimental_feature) {
-                        *slot_mut += " ";
-                        *slot_mut += experimental_feature;
-                    }
-                }
-            },
-            Entry::Vacant(slot) => {
-                let _ = slot.insert(experimental_features.join(" "));
-            },
-        };
 
-        // https://github.com/DeterminateSystems/nix-installer/issues/449#issuecomment-1551782281
-        #[cfg(not(target_os = "macos"))]
-        settings.insert("auto-optimise-store".to_string(), "true".to_string());
+        let mut nix_conf_insert_settings = Vec::default();
+        nix_conf_insert_settings.push("!include ./defaults.conf".into());
```

Comment:
We should name this file something like `detsys-nix-installer-defaults.conf` so it's obvious where it came from without needing to look at the contents.

Reply:
Also we probably want it to be `include`, not `!include` -- `!include` means "if it doesn't exist, that's fine" to Nix.